docs-tech/bakery-architecture.md

```diff
@@ -52,3 +52,74 @@ and are pure computations, which traditionally are called "sieves".
 A library to be included within the consumers of bakery, to ease the integration of old and new baker users. 
 It exposes the traditional baker java and scala api, but within its implementation it requests through http 
 to the exposed baker clusters. One instance of the client is created per baker cluster.
+
+
+# Bakery Kubernetes Implementation
+
+![Kubernetes diagram](./Bakery%20Kubernetes%20Architecture.png)
+
+## Bakery Controller
+
+The Bakery Controller follows Kubernete's general architecture pattern called [Controller Pattern](https://kubernetes.io/docs/concepts/architecture/controller/)
+It is deployed as a single pod which uses Kubernete's API to watch the Baker and Interaction CRDs, which have the
+specification of the Baker clusters and Interactions to be deployed. The Baker Controller constantly watches for creation,
+changes and deletions of such CRDs, so that the state of the namespace always matches the textual specifications.
+
+The Bakery Controller also creates Kubernetes ConfigMaps which we call "Manifests", which are outcome of the creation of
+a CRD, these are config maps used as internal mechanisms to expose semantic data about the deployed components, more of
+that in the following section.
+
+## Baker and Interaction CRDs
+
+The 2 CRDs that we have contain all the requirements of the components to be deployed, like the amount of cluster nodes 
+/ replicas, the encoded recipes to be added to the bakers, or the docker images to be used for interactions.
+
+The Bakery Controller can be configured to use ConfigMaps instead of CRDs in case you are not able to use CRDs in your cluster
+by adding `bakery-controller.use-crds = false` to the `application.conf` of the controller.
+
+An example of CRD can be found [here for a baker](../bakery-integration-tests/src/test/resources/kubernetes/crd/baker-webshop.yaml)
+and [here for an interaction](../bakery-integration-tests/src/test/resources/kubernetes/crd/interactions-example.yaml)
+
+Also the specification of the CRDs can be found [here for Baker](../bakery-integration-tests/src/test/resources/kubernetes/crd/crd-baker.yaml)
+and [here for Interaction](../bakery-integration-tests/src/test/resources/kubernetes/crd/crd-interaction.yaml)
+
+### Baker Manifests
```

Comment:
Here and 3 sections below: just one long sentence per section. 
Splitting it to shorter sentences and avoiding passive voice where possible would improve readability.